app/src/atoms/InputField/index.tsx

```diff
@@ -141,38 +212,55 @@ function Input(props: InputFieldProps): JSX.Element {
     }
   `
 
+  const UNITS_STYLE = css`
+    color: ${props.disabled ? COLORS.grey40 : COLORS.grey50};
+    font-size: ${TYPOGRAPHY.fontSizeLabel};
+    font-weight: ${TYPOGRAPHY.fontWeightSemiBold};
+    line-height: ${TYPOGRAPHY.lineHeight12};
+    align-text: ${TEXT_ALIGN_RIGHT};
+    @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+      color: ${props.disabled ? COLORS.grey40 : COLORS.grey50};
+      font-size: ${TYPOGRAPHY.fontSize22};
+      font-weight: ${TYPOGRAPHY.fontWeightRegular};
+      line-height: ${TYPOGRAPHY.lineHeight28};
+      justify-content: ${textAlign};
+    }
+  `
+
   return (
-    <Flex width="100%" flexDirection={DIRECTION_COLUMN}>
-      <Flex css={INPUT_FIELD}>
-        <input
-          {...props}
-          data-testid={props.id}
-          value={value}
-          placeholder={placeHolder}
-        />
-        {props.units != null && (
-          <Flex
-            display={DISPLAY_INLINE_BLOCK}
-            textAlign={TEXT_ALIGN_RIGHT}
-            alignSelf={ALIGN_CENTER}
-            color={props.disabled ? COLORS.grey40 : COLORS.grey50}
-            fontSize={TYPOGRAPHY.fontSizeLabel}
-          >
-            {props.units}
-          </Flex>
-        )}
-      </Flex>
-      <Flex
-        color={COLORS.grey50}
-        fontSize={TYPOGRAPHY.fontSizeLabel}
-        paddingTop={SPACING.spacing4}
-        flexDirection={DIRECTION_COLUMN}
-      >
-        <Flex css={FORM_BOTTOM_SPACE_STYLE}>{props.caption}</Flex>
-        {props.secondaryCaption != null ? (
-          <Flex css={FORM_BOTTOM_SPACE_STYLE}>{props.secondaryCaption}</Flex>
-        ) : null}
-        <Flex css={ERROR_TEXT_STYLE}>{props.error}</Flex>
+    <Flex flexDirection={DIRECTION_COLUMN} width="100%">
+      {props.title != null ? (
+        <Flex css={TITLE_STYLE}>{props.title}</Flex>
+      ) : null}
+      <Flex width="100%" flexDirection={DIRECTION_COLUMN} css={OUTER_CSS}>
+        <Flex
+          css={INPUT_FIELD}
+          alignItems={ALIGN_CENTER}
+          as="label"
+          for={inputProps.id}
+        >
+          <input
+            {...inputProps}
+            data-testid={props.id}
+            value={value}
+            placeholder={placeHolder}
+          />
+          {props.units != null ? (
+            <Flex css={UNITS_STYLE}>{props.units}</Flex>
+          ) : null}
+        </Flex>
+        <Flex
+          color={COLORS.grey60}
+          fontSize={TYPOGRAPHY.fontSizeLabel}
+          paddingTop={SPACING.spacing4}
+          flexDirection={DIRECTION_COLUMN}
+        >
+          <Flex css={FORM_BOTTOM_SPACE_STYLE}>{props.caption}</Flex>
```

Comment:
from caption's type adding a condition would be safe?

```suggestion
{props.caption != null ? (
          <Flex css={FORM_BOTTOM_SPACE_STYLE}>{props.caption}</Flex>
          ) : null}
```